api/src/lang/machine.rs

```diff
@@ -117,10 +117,32 @@ impl MachineState {
         let mut input = env.input.clone();
         input.reverse(); // Reverse the vec so we can pop off the values in order
         Self {
-            stacks: Stacks::new(&env),
             input,
             output: Vec::new(),
-            workspace: 0,
+            registers: iter::repeat(0)
+                .take(usize::try_from(env.num_registers).unwrap())
+                .collect(),
+            stacks: Stacks::new(&env),
```

Comment:
still confused on this iter.repeat(0) stuff

Reply:
Repeat just infinitely repeats whatever value you give it, so in this case it's repeating 0 once per register.

Reply:
o that makes more sense